source/Plugins.cpp

```diff
@@ -164,3 +187,77 @@ void Plugins::TogglePlugin(const string &name)
 	auto *plugin = plugins.Get(name);
 	plugin->currentState = !plugin->currentState;
 }
+
+
+
+future<void> Plugins::Install(const InstallData &installData, bool guarded)
+{
+	oldNetworkActivity = true;
+	if(!guarded)
+	{
+		lock_guard<mutex> guard(activePluginsMutex);
+		if(!activePlugins.insert(installData.name).second)
+			return future<void>();
+
+		// Create a new entry for the plugin.
```

Comment:
Shouldn't the new entry only be created if installation succeeds?

Reply:
yea, did that